torch/distributed/tensor/_api.py

```diff
@@ -286,6 +286,11 @@ def __new__(
         r._local_tensor = local_tensor
         return r
 
+    from torch._export.wrappers import _mark_constructor_exportable_experimental
+    @_mark_constructor_exportable_experimental
```

Comment:
on the one hand, you could just put the decorator over DTensor's `__new__`, since they actually initialize their subclass contents in the __new__.

I guess adding an init here and putting it here is better for consistency though, in case people look at DTensor to copy code from

Reply:
I feel it is better to make the constructor work for just one case (either __new__ or __init__) because both have different semantics and concurrently supporting them feel bit fragile as the current logic is very specific to the implementaitons of them.